src/GraphQL.Tests/Utilities/Visitors/PatternMatchingVisitorTests.cs

```diff
@@ -57,4 +57,21 @@ public static string Hello(
             string arg)
             => arg;
     }
+
+    [Theory]
+    [InlineData(null, "Pattern directive 'regex' argument must have non-null value")]
+    [InlineData(123, "Pattern directive 'regex' argument must be of 'string' type")]
+    public void InvalidRegexValue(object? argumentValue, string message)
+    {
+        var query = new ObjectGraphType { Name = "Query" };
+        query.Field<string>("Hello")
+            .Argument<StringGraphType>("arg", argument => argument.ApplyDirective("pattern", "regex", argumentValue))
```

Comment:
Btw, I noticed that `argumentValue` in `ApplyDerective` is defined as nullable, but in `DirectiveAttribute` as non-nullable...